### PR TITLE
add site url checker

### LINKED ISSUE #37: Hey does this script support subdomains?
It always fail whenever I try to index subdomain, I setup a URL prefix for the subdomain.

```
🔎 Processing site: sc-domain:sub.example.com
🔐 This service account doesn't have access to this site.
❌ No sitemaps found, add them to Google Search Console and try again.
```

### PR DESCRIPTION
**Pull request by antoinekm**
**What did I change?**

Before running the rest of the script, we check that the user has access to the requested site, and if he's made a mistake, we correct the URL.

**Why did I change it?**

Many users make a mistake, or encounter an error related to this when it's just a matter of understanding, so we needed a solution to fix this problem automatically. (close: #37)

### CODE REVIEW

**Comment by AntoineKM** [comment]
Enregistrement de l’écran 2024-03-18 à 17.07.46

**Comment by antoinekm** [comment]
https://github.com/goenning/google-indexing-script/assets/54948363/40d03e3c-b83b-4cef-85b2-41eacab5ced3

File: src/shared/gsc.ts [hunk 1]
```diff
@@ -22,6 +23,126 @@ export function convertToFilePath(path: string) {
...
+  return sitesBody.siteEntry.map((x) => x.siteUrl);
+}
+
+/**
+ * Checks if the site URL is valid and accessible by the service account.
+ * @param accessToken - The access token for authentication.
+ * @param siteUrl - The URL of the site to check.
+ * @returns The corrected URL if found, otherwise the original site URL.
+ */
+export async function checkSiteUrl(
```
Comment (goenning) [inline src/shared/gsc.ts]: this function is very hard to read 😅

maybe we could start this function by converting to all 3 possible format/values and then check which one matches. 

And then just carry on with the process, I think notifications like this are unnecessary, most people won't care to change if they know the script works anyway
```
console.warn(`🚨 Found HTTPS version of the site, please next time use this format instead: ${correctUrl}`);
```

**Comment by goenning** [comment]
that was fast! We could probably simplify the readme a bit and just tell the user to use `gis domain.com` as the script will figure out which format to use

**Comment by antoinekm** [comment]
Yes, it should be better now 😁

**Pull request merged**